usr/sbin/p11sak/p11sak.c

```diff
@@ -3956,6 +4117,74 @@ static void print_byte_array_attr(const char *attr, const CK_ATTRIBUTE *val,
     }
 }
 
+static void print_x509_name_attr(const char *attr, const CK_ATTRIBUTE *val,
+                                 int indent, bool sensitive)
+{
+    X509_NAME *name = NULL;
+    const unsigned char *tmp_ptr;
+
+    if (sensitive) {
+        printf("%*s%s: [sensitive]\n", indent, "", attr);
+        return;
+    } else if (val->ulValueLen == 0) {
+        printf("%*s%s: [no value]\n", indent, "", attr);
+        return;
+    }
+
+    tmp_ptr = (const unsigned char *)val->pValue;
+    name = d2i_X509_NAME(NULL, &tmp_ptr, val->ulValueLen);
+    if (name != NULL) {
+        char *oneline = X509_NAME_oneline(name, NULL, 0);
```

Comment:
What if X509_NAME_oneline() returns NULL ?

Reply:
Same thought as for the comment below: the function returns an internal pointer. It "should" not be NULL.

Reply:
OK, `X509_get_subject_name()` really just returns an internal pointer so it is probably save to assume it never return NULL, but `X509_NAME_oneline()` is way more than just returning an internal pointer, see https://github.com/openssl/openssl/blob/7c729851d169f30d9e0c0ad6e7c1cf6cefb37935/crypto/x509/x509_obj.c#L25
So its a about 150 lines function with lots of possible NULL returns....

Reply:
Done.